osu.Game/Screens/Play/Player.cs

```diff
@@ -412,7 +412,7 @@ public void Restart()
 
         private ScheduledDelegate completionProgressDelegate;
 
-        private void onCompletion()
+        private void updateCompletionState(ValueChangedEvent<bool> completionState)
```

Comment:
this is going to trigger multiple times. you aren't cancelling or checking the value is true even.

Reply:
right, I'll close this. you can review #8800 directly instead.